modules/quant_purger/src/StackMiddleware/UrlRegistrar.php

```diff
@@ -2,112 +2,127 @@
 
 namespace Drupal\quant_purger\StackMiddleware;
 
-use Drupal\Core\Cache\CacheableResponseInterface;
 use Drupal\Core\Config\ConfigFactoryInterface;
 use Drupal\quant_purger\TrafficRegistryInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\HttpKernelInterface;
 
-/**
- * Collects URLs that Quant has requested.
- */
-class UrlRegistrar implements HttpKernelInterface {
+$reflection = new \ReflectionClass(HttpKernelInterface::class);
+$method = $reflection->getMethod('handle');
 
-  /**
-   * The HTTP kernel object.
-   *
-   * @var \Symfony\Component\HttpKernel\HttpKernelInterface
-   */
-  protected $httpKernel;
+// In Drupal10+ HttpKernelInterface has been updated with
+// Symfony to have a different signature.
+//
+// @see https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/HttpKernelInterface.php.
+if (empty($method->getReturnType())) {
```

Comment:
Now with more reflection — d9+d10 support in the same place.

Reply:
I diffed the two classes:

```
41c41
<     public function handle(Request $request, $type = self::MASTER_REQUEST, $catch = TRUE) {
---
>     public function handle(Request $request, int $type = self::MAIN_REQUEST, bool $catch = TRUE): Response {
42a43
>       var_dump('d10');
```

I see this in 4.4 and lower:

https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/HttpKernel/HttpKernelInterface.php

```
    public function handle(Request $request, $type = self::MASTER_REQUEST, $catch = true);
```

I see this in 5.0+

```
    public function handle(Request $request, int $type = self::MAIN_REQUEST, bool $catch = true): Response;
```

so it looks like this matches the signatures... 

Need to remove the var_dump :)

Reply:
I see the response was added in 9.3:

https://www.drupal.org/project/drupal/issues/3232082

not sure about the rest.

Reply:
Fixed up the var_dump — should be good to go :D